### ISSUE TITLE
Can't run my signed & notarized Electron app in macOS Catalina

### ISSUE THREAD

**Issue by DTLTAKE** [electron/electron#22884]
Hi all,

After some trouble, I have managed to get my Electron app signed and notarized for distribution in macOS. However, when testing the .dmg installer in another computer (running macOS Catalina), it fails to run - the splash screen appears at the beggining and then the app just closes.

My app is built on Electron 4.0.6 and uses both Python (backend) and javascript (frontend), which communicate over a websocket (which is initialised in the splashscreen). My app reads and writes files from a folder in Documents (config files) and also reads and saves files through user input. Furthermore, my app performs acquisitions over Bluetooth and USB, and uses the user's location.

I have been using the following entitlements (entitlements.mac.plist) for signing entitlements and entitlementsInherit:
```
<?xml version="1.0" encoding="UTF-8"?>  
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">  
<plist version="1.0">  
<dict>  
    <key>com.apple.security.cs.allow-jit</key><true/>  
    <key>com.apple.security.cs.allow-unsigned-executable-memory</key><true/>  
    <key>com.apple.security.cs.disable-library-validation</key><true/>  
    <key>com.apple.security.automation.apple-events</key><true/>  
    <key>com.apple.security.temporary-exception.files.absolute-path.read-only</key><true/>  
    <key>com.apple.security.temporary-exception.files.absolute-path.read-write</key><true/>  
    <key>com.apple.security.network.client</key><true/>  
    <key>com.apple.security.network.server</key><true/>  
    <key>com.apple.security.assets.movies.read-write</key><true/>  
    <key>com.apple.security.assets.music.read-write</key><true/>  
    <key>com.apple.security.assets.pictures.read-write</key><true/>  
    <key>com.apple.security.files.downloads.read-write</key><true/>  
    <key>com.apple.security.files.user-selected.read-write</key><true/>  
    <key>com.apple.security.device.usb</key><true/>  
    <key>com.apple.security.device.bluetooth</key><true/>  
    <key>com.apple.security.personal-information.location</key><true/>  
</dict>  
</plist>  
```

To run the signing I use the following command:
`./node_modules/.bin/electron-osx-sign dist/MyApp.app --no-gatekeeper-assess --identity="Developer ID Application: COMPANY (#######)" --entitlements=dist/entitlements.mac.plist --entitlementsInherit=dist/entitlements.mac.plist  `

Afterwards, I sign all the files with extension .so, .dylib, .a and .o using the following XCode command:
```
codesign --sign "Developer ID Application: COMAPNY (########)" \  
                 --entitlements dist/entitlements.mac.plist \  
                 --deep filename \  
                 --force \  
                 --timestamp \  
                 --options runtime; 
```

The log I obtain when running my app (which crashes) can be found here: https://i.imgur.com/zz38t1b.png

Any idea of what is wrong? I can't manage to make it work.

**Comment by MarshallOfSound** [comment]
@DTLTAKE You don't want to run `codesign` manually.  Just run `electron-osx-sign` with `--hardened-runtime`

```
./node_modules/.bin/electron-osx-sign dist/MyApp.app --no-gatekeeper-assess --identity="Developer ID Application: COMPANY (#######)" --entitlements=dist/entitlements.mac.plist --entitlementsInherit=dist/entitlements.mac.plist --hardened-runtime
```

Then you use [`electron-notarize`](https://github.com/electron/electron-notarize) to notarize your app.

**Comment by DTLTAKE** [comment]
@MarshallOfSound thanks for your reply.

I have tried your suggestion of signing the app with `--hardened-runtime` enabled, but doing so makes my app to show a blank white screen when running.
I have also tried adding the flag `--type=distribution` but it didn't work. I also tried adding the `<key>com.apple.security.cs.disable-executable-page-protection</key><true/>` entitlement to my entitlements (in the first post), but with no success. Updating electron to version 4.2.12 also changed nothing.

Any clues on how to solve this issue?
